### ISSUE TITLE
Suggestion: Extend setVariable to allow 'output' variables to be created

### ISSUE THREAD

**Issue by chris-codeflow** [microsoft/azure-pipelines-task-lib#688]
### Environment
azure-pipelines-task-lib version: n/a

### Issue Description
The `setVariable` method only allows job-scoped variables to be created.  Consider adding an `isOutput: boolean` parameter to the method, or create a new method to allow output variables (i.e. adding the [`isOutput=true`](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#set-a-multi-job-output-variable) property to the `task.setvariable` logging command) to be created.

**Comment by johnnyreilly** [comment]
Hi, 

I've just been bumping on this as well.  From the looks of it the change would be made here: 

https://github.com/microsoft/azure-pipelines-task-lib/blob/master/node/task.ts#L162

Is there a chance we could get this added?

**Comment by johnnyreilly** [comment]
In the interim I've come up with a workaround which can be used:

```ts
import * as tl from 'azure-pipelines-task-lib/task';
import * as tcm from 'azure-pipelines-task-lib/taskcommand';
import * as os from 'os';

/**
 * Sets a variable which will be output as well.
 *
 * @param     name    name of the variable to set
 * @param     val     value to set
 * @param     secret  whether variable is secret.  Multi-line secrets are not allowed.  Optional, defaults to false
 * @returns   void
 */
export function setOutputVariable(name: string, val: string, secret = false): void {
    // use the implementation of setVariable to set all the internals,
    // then subsequently set the output variable manually
    tl.setVariable(name, val, secret);

    const varValue = val || '';

    // write the command
    // see https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#set-a-multi-job-output-variable
    _command(
        'task.setvariable',
        { variable: name || '', isOutput: 'true', issecret: (secret || false).toString() },
        varValue
    );
}

const _outStream = process.stdout;

function _writeLine(str: string): void {
    _outStream.write(str + os.EOL);
}

function _command(command: string, properties: any, message: string) {
    const taskCmd = new tcm.TaskCommand(command, properties, message);
    _writeLine(taskCmd.toString());
}
```

**Comment by chris-codeflow** [comment]
@damccorm .  I mentioned in a previous post that I would have raised a PR myself but I wasn't sure of the contribution guidelines.   The current contribution guidelines include "Features (things that add new or improved functionality) may be accepted, but will need to first be approved in the form of a suggestion issue.".

This is what I did.  I raised my suggestion issue and was expecting it to be approved (with a label or comment) so I could start work on it.
